flocker/node/agents/cinder.py

```diff
@@ -530,7 +530,7 @@ def list_volumes(self):
         http://docs.rackspace.com/cbs/api/v1.0/cbs-devguide/content/GET_getVolumesDetail_v1__tenant_id__volumes_detail_volumes.html
         """
         flocker_volumes = []
-        for cinder_volume in self.cinder_volume_manager.list(detailed=False):
+        for cinder_volume in self.cinder_volume_manager.list(detailed=True):
```

Comment:
Yes that's right. 

* Cinder v1 API volume list includes metadata: http://developer.openstack.org/api-ref-blockstorage-v1.html#listVolumes
* Cinder v2 doesn't: http://developer.openstack.org/api-ref-blockstorage-v2.html#listVolumes


That's what led to this bug. I tested against Rackspace and saw the tests pass. 
On OpenStack the detailed list doesn't include the attachment info.